src/main/java/net/imagej/ops/descriptor3d/MarchingCubes.java

```diff
@@ -0,0 +1,497 @@
+package net.imagej.ops.descriptor3d;
+
+import net.imagej.ops.AbstractFunctionOp;
+import net.imagej.ops.FunctionOp;
+import net.imagej.ops.Op;
+import net.imagej.ops.OpService;
+import net.imagej.ops.Ops.Descriptor3D;
+import net.imagej.ops.Ops.Descriptor3D.Polygonize;
+import net.imglib2.Cursor;
+import net.imglib2.FinalInterval;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.logic.BoolType;
+import net.imglib2.view.ExtendedRandomAccessibleInterval;
+import net.imglib2.view.Views;
+
+import org.scijava.ItemIO;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * This is a marching cubes implementation. It is inspired by
+ * Paul Bourke's (http://paulbourke.net/geometry/polygonise/) implementation. 
+ * Especially the lookup tables are from his implementation. 
+ * 
+ * @author Tim-Oliver Buchholz, University of Konstanz
+ *
+ * @param <T> BooleanType
+ */
+@Plugin(type = Op.class, name = Descriptor3D.Polygonize.NAME)
+public class MarchingCubes<T extends BooleanType<T>> extends
+		AbstractFunctionOp<RandomAccessibleInterval<T>, DefaultFacets>
+		implements Polygonize, FunctionOp<RandomAccessibleInterval<T>, DefaultFacets> {
```

Comment:
Where does the `Polygonize` interface stem from? Better `Interface` + `Default` as usual.